rpqueue/__init__.py

```diff
@@ -595,7 +595,10 @@ def __call__(self, *args, **kwargs):
             if want_status:
                 conn.delete(k)
             raise
-        del CURRENT_TASK.task
+        try:
+            del CURRENT_TASK.task
+        except AttributeError:
+            pass
```

Comment:
Any info on why we do this?

Reply:
This is coming from the original library. The author did some cleanup. I borrowed code.

Reply:
This definitely needs some sort comment to explain this special handling.

Reply:
+1 to at least adding a comment here.